skycoin/skywire
# WebSocket API for updater

Did you run `make format && make check`? Not yet

Fixes #413 

Changes:	
- Add WebSocket API for updating visor
- Add WebSocket API for updating hypervisor
- Add draft support of WebSocket API to the frontend for testing and debugging purposes, the code likely needs to be modified or rewritten (/cc @Senyoret1) 
- Send update progress to frontend in WebSocket API
- If called multiple times, updater WebSocket endpoints return current progress instead of starting a new updating process
- Fix `.goreleaser.yml`

How to test this PR:
1. Change `VERSION` variable in `Makefile` to `v0.1.0`
2. Change `channel: 'stable'` to `channel: 'testing'` at `./static/skywire-manager-src/src/app/services/node.service.ts#213`
3. Run `make build-ui`
4. Run `make integration-build`
5. Run `make integration-run-generic`
6. Open hypervisor web UI
7. Update a visor
8. The visor should become updated with no errors
9. To test hypervisor updating, change `visors/${nodeKey}/update/ws` to `update/ws` at `./static/skywire-manager-src/src/app/services/node.service.ts#216` and follow steps 3-8, instead of visor, hypervisor will be updated
10. To test returning current progress on multiple WebSocket endpoint calls, start an update, refresh the browser page and run the update again. Instead of starting a new updating, the progress of the old one should be shown.
